### PR TITLE
Welcome Relay

### PR DESCRIPTION
**Pull request by doylemark**
# Why
Adds create & deploy relay commands. These are useful for managing your relay's in CI. 
# How
This is a WIP pr ahead of a new release

### CODE REVIEW

File: crates/ev-cli/src/relay/mod.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+use std::{fs, io};
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum RelayConfigError {
+    #[error(
+        "No relay.json file found in this directory, specify a relay config file \
+        with the --file flag. Or create a relay with ev relay create."
+    )]
+    ConfigNotFound(String),
```
Comment (lfarrel6) [inline crates/ev-cli/src/relay/mod.rs:11]: Should the path be included in the message?

File: crates/ev-cli/src/commands/relay/create.rs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+            CreateError::WriteError(_) => crate::errors::CANTCREAT,
+            _ => crate::errors::GENERAL,
+        }
+    }
+}
+
+#[derive(strum_macros::Display, Debug)]
+pub enum CreateMessage {
+    #[strum(to_string = "Relay configuration saved to file")]
+    FileWritten,
```
Comment (lfarrel6) [inline crates/ev-cli/src/commands/relay/create.rs]: Can we include the file path in the message here to make it more obvious?

File: crates/ev-cli/src/commands/relay/create.rs [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    pub force: bool,
+}
+
+#[derive(Error, Debug)]
+pub enum CreateError {
+    #[error(
+        "A Relay configuration file already exists at the path: {0}, use the --force parameter to overwrite the existing file"
+    )]
+    FileAlreadyExists(String),
+    #[error("An error occured while writing the Relay configuration file:\n{0}]")]
```
Comment (lfarrel6) [inline crates/ev-cli/src/commands/relay/create.rs]: We're not using the line break after our message in `RelayConfigError`, should be consistent (don't think the line break is necessary)

File: crates/ev-cli/src/theme.rs [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+    /// Unpicked item in sort prefix value and style
+    pub unpicked_item_prefix: StyledObject<String>,
+    /// Show the selections from certain prompts inline
+    pub inline_selections: bool,
+}
+
+impl Default for CliTheme {
+    fn default() -> CliTheme {
+        let report = crate::tty::AttyReport::default();
+        if !report.stdout | has_color_env_vars() {
```
Comment (lfarrel6) [inline crates/ev-cli/src/theme.rs]:
```suggestion
        if !report.stdout || has_color_env_vars() {
```

File: crates/ev-cli/src/theme.rs [hunk 5]
```diff
@@ -0,0 +1,373 @@
...
+            )?;
+        }
+
+        match default {
+            Some(default) => write!(
+                f,
+                "{} {} ",
+                self.hint_style.apply_to(&format!("({})", default)),
+                &self.prompt_suffix
+            ),
```
Comment (lfarrel6) [inline crates/ev-cli/src/theme.rs]: default is an [unstable keyword in rust](https://github.com/rust-lang/rust/issues/31844), should avoid using in vars:
```suggestion
            Some(default_val) => write!(
                f,
                "{} {} ",
                self.hint_style.apply_to(&format!("({})", default_val)),
                &self.prompt_suffix
            ),
```

File: crates/ev-cli/src/theme.rs [hunk 6]
```diff
@@ -0,0 +1,373 @@
...
+                &self.unchecked_item_prefix,
+                self.active_item_style.apply_to(text),
+            ),
+            (false, false) => (
+                &self.unchecked_item_prefix,
+                self.inactive_item_style.apply_to(text),
+            ),
+        };
+
+        write!(f, "{} {}", details.0, details.1)
```
Comment (lfarrel6) [inline crates/ev-cli/src/theme.rs]: This seems needlessly verbose
```suggestion
        let prefix = if checked { 
          self.checked_item_prefix 
        } else { 
          self.unchecked_item_prefix 
        };
        
        let prompt_item = if active { 
          self.active_item_style.apply_to(text) 
        } else { 
          self.inactive_item_style.apply_to(text) 
        };
       
        write!(f, "{} {}", prefix, prompt_item)
```

File: crates/ev-cli/src/theme.rs [hunk 7]
```diff
@@ -0,0 +1,373 @@
...
+            ),
+            (false, true) => (
+                &self.unpicked_item_prefix,
+                self.active_item_style.apply_to(text),
+            ),
+            (_, false) => (
+                &self.unpicked_item_prefix,
+                self.inactive_item_style.apply_to(text),
+            ),
+        };
```
Comment (lfarrel6) [inline crates/ev-cli/src/theme.rs]: same as above

**Review by lfarrel6 — COMMENTED** [review]
some small nit picks, but looks good

File: crates/ev-cli/src/theme.rs (hunk 5 above)
Comment (doylemark) [reply crates/ev-cli/src/theme.rs]: til

**Review by lfarrel6 — APPROVED** [review]

**Pull request merged**